infrastructure/deployment/lambda-cfn.yaml

```diff
@@ -0,0 +1,98 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: Differential service configuration
+
+Parameters:
+  ClusterId:
+    Type: String
+    Description: Differential Cluster Id
+
+  ServiceName:
+    Type: String
+    Description: Differential Service name
+
+  DeploymentId:
+    Type: String
+    Description: Differential Deployment Id
+
+  FunctionName:
+    Type: String
+    Description: The Lambda function name
+
+  Runtime:
+    Type: String
+    Description: The Lambda runtime
+    Default: nodejs20.x
+
+  UploadBucketName:
+    Type: String
+    Description: The S3 Upload bucket name
+
+  UploadBucketKey:
+    Type: String
+    Description: The S3 Upload bucket key
+
+  APISecret:
+    Type: String
+    NoEcho: true
+    Description: Differential API key for handler
+
+  Handler:
+    Type: String
+    Description: The name of the method that Lambda calls to run the function
+    Default: "differential-index.handler"
+
+  Timeout:
+    Type: String
+    Description: The Lambda function timeout
+    Default: "60"
+
+  ReservedConcurrency:
+    Type: String
+    Description: The Lambda function reserved concurrency
+    Default: "1"
+
+  ExecutionRole:
+    Type: String
+    Description: Lambda execution role
+    Default: ""
+
+Conditions:
+  IsExecutionRoleEmpty:
+    "Fn::Equals":
+      - !Ref "ExecutionRole"
+      - ""
+
+Resources:
+  DifferentialService:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Ref FunctionName
+      Handler: !Ref Handler
+      Role:
+        'Fn::If':
+          - IsExecutionRoleEmpty
+          - !Sub 'arn:aws:iam::${AWS::AccountId}:role/DeploymentLambdaRuntimeRole'
```

Comment:
Is this a role that was already created in AWS? 

Would it be worth creating a role per lambda in this template and associating that with the function?

Reply:
This is created as part of [bootstrapping the environment](https://github.com/differentialhq/differential/blob/main/infrastructure/cfn.yaml#L52).

The `control-plane` can override these via [parameters](https://github.com/differentialhq/differential/pull/165/files#diff-dba2bfe46dd179980d16b3d3af6a03b096c47a826cda311d23fa9786eab59c44R160).

But yeah, we could easily enough create a new role for each deployment.